sn_logging/src/metrics.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use serde::Serialize;
+use std::time::Duration;
+use sysinfo::{self, CpuExt, NetworkExt, Pid, PidExt, ProcessExt, System, SystemExt};
+use tracing::trace;
+
+const UPDATE_INTERVAL: Duration = Duration::from_secs(5);
```

Comment:
will the 5s interval being too frequent ?
especially with a public testnet ?

Reply:
I think 5s would be optimal resolution for us to correlate the metrics with any anomaly that we might be interested in? Anything higher, it would be hard for us to pinpoint what caused the spikes in mem/cpu.
Currently, we store a max of 1000 logs each with 5000 lines, so that should be plenty of space.